Compile Error Melodic due to tf change

A change in ```tf.h``` here: https://github.com/ros/geometry/pull/163
causes the following compile error on Ubuntu 18.04 and Melodic:

```
Errors     << amcl:make /home/amoriarty/melodic/logs/amcl/build.make.000.log                                                                                                                               
/home/amoriarty/melodic/src/navigation/amcl/src/amcl_node.cpp: In member function ‘tf2_ros::Buffer& AmclNode::TransformListenerWrapper::getBuffer()’:
/home/amoriarty/melodic/src/navigation/amcl/src/amcl_node.cpp:130:51: error: ‘tf2_buffer_’ was not declared in this scope
       inline tf2_ros::Buffer &getBuffer() {return tf2_buffer_;}
                                                   ^~~~~~~~~~~
/home/amoriarty/melodic/src/navigation/amcl/src/amcl_node.cpp:130:51: note: suggested alternative: ‘tf2_buffer_ptr_’
       inline tf2_ros::Buffer &getBuffer() {return tf2_buffer_;}
                                                   ^~~~~~~~~~~
                                                   tf2_buffer_ptr_
make[2]: *** [CMakeFiles/amcl.dir/src/amcl_node.cpp.o] Error 1
make[1]: *** [CMakeFiles/amcl.dir/all] Error 2
make: *** [all] Error 2

```
The simple gcc suggestion to replace ```tf2_buffer_``` with ```tf2_buffer_ptr_``` doesn't fix the problem, just gets the next error msg:

```
Errors     << amcl:make /home/amoriarty/melodic/logs/amcl/build.make.001.log                                                                                                                               
/home/amoriarty/melodic/src/navigation/amcl/src/amcl_node.cpp: In member function ‘tf2_ros::Buffer& AmclNode::TransformListenerWrapper::getBuffer()’:
/home/amoriarty/melodic/src/navigation/amcl/src/amcl_node.cpp:130:51: error: invalid initialization of reference of type ‘tf2_ros::Buffer&’ from expression of type ‘std::shared_ptr<tf2_ros::Buffer>’
       inline tf2_ros::Buffer &getBuffer() {return tf2_buffer_ptr_;}
                                                   ^~~~~~~~~~~~~~~
```

I was using the **lunar** branch earlier as it was the most up to date and was working with melodic, but came back from holidays and it fails to compile.

I just brought this up with @aaronhoy, I think to open a pull request to fix this issue requires a **melodic-devel** branch

--- a/amcl/src/amcl_node.cpp
+++ b/amcl/src/amcl_node.cpp
@@ -127,7 +127,7 @@ class AmclNode
     // Use a child class to get access to tf2::Buffer class inside of tf_
     struct TransformListenerWrapper : public tf::TransformListener
     {
-      inline tf2_ros::Buffer &getBuffer() {return tf2_buffer_;}
+      inline std::shared_ptr<tf2_ros::Buffer> getBuffer() {return tf2_buffer_ptr_;}
     };
 
     TransformListenerWrapper* tf_;
@@ -649,7 +649,7 @@ void AmclNode::runFromBag(const std::string &in_bag_fn)
       tf_pub.publish(msg);
       for (size_t ii=0; ii<tf_msg->transforms.size(); ++ii)
       {
-        tf_->getBuffer().setTransform(tf_msg->transforms[ii], "rosbag_authority");
+        tf_->getBuffer()->setTransform(tf_msg->transforms[ii], "rosbag_authority");
       }
       continue;
     }

